### PR TITLE
[FABCN-373] Compatibility Matrix

### PR DESCRIPTION
**Pull request by mbwhite**
Signed-off-by: Matthew B White <whitemat@uk.ibm.com>
Change-Id: Ia536ed66daf857679adf9f858b35d206a4f1007f

### CODE REVIEW

File: SUPPORT.md [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+## Summary of Compatibility
+
+Summary of the Fabric versions and the default node runtimes they use to host chaincode and contracts; along side the version of the node modules and which runtimes they are compatible with.
+
+|                       | Fabric 1.4 (node v8) | Fabric v2.0 (node v12) |
+| --------------------- | -------------------- | ---------------------- |
+| node modules **v1.4** |         Yes             |      Yes                  |
+| node modules **v2.0** |         No             |          Yes              |
+
+This is the case by default, however the docker image used to host the chaincode and contracts can be altered by configuring the Fabric Peer.  Set the environment variable `CORE_CHAINCODE_NODE_RUNTIME` on the peer to the name of the docker image.
```
Comment (awjh-ibm) [inline SUPPORT.md]: Could this have an example docker image name for clarity, I had to read it a couple of times to work out what it wanted?

File: SUPPORT.md [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+- When starting a chaincode container to run a Smart Contract the version of the runtime that is used is determined by these factors:
+
+Fabric v1.4.2, and Fabric v2.0.0 will by default start up a docker image, that is defined by the version of Fabric in use.
+
+
+
+With Fabric v2.0.0, the chaincode container can be configured to be started by other means, and not the Peer. In this case, the environment used is not in the control of Fabric.
+ 
+
+npm modules that are produced are the `fabric-contract-api`, `fabric-shim`, `fabric-shim-crypto` & `fabric-shim-api`
```
Comment (awjh-ibm) [inline SUPPORT.md]: Remove the the?

File: SUPPORT.md [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+
+### Supported Runtimes
+
+v1.4.x and v2.0.0 npm modules are supported running in node12.13 with the x86_64 architecture. Later node 12 versions are supported but are not tested.
+
+v1.4.x npm modules are supported running in node8.16.1 with the  x86_64 architecture. Later node 8 versions are supported but are not tested
```
Comment (awjh-ibm) [inline SUPPORT.md]: Extra space

File: SUPPORT.md [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+
+When using Fabric 2.0.0, the default docker image that is used to run the Node chaincode is node:12.13.0-alpine 
+
+Note: With the default docker image used by Fabric 2.0.0 the packaged code will be installed with npm. If a `package-lock.json` or a `npm-shrinkwrap.json` file is present, `npm ci --only=production` will be used. Otherwise `npm install --production` will be used. 
+
+When using Fabric 1.4.4, the docker image that is used to run the Node chaincode is node v8.16.1. It is installed with npm install --production
+
+### Supported Runtime communication with the Peer
+ 
+Subject to a suitable runtime environment, the 1.4.4 npm modules   and 2.0.0 npm modules can used to communicate with a Fabric 2.0.0 or 1.4.4 Peer - with the level of functionality that is implied by the Fabric version in use.
```
Comment (awjh-ibm) [inline SUPPORT.md]: Extra space

File: SUPPORT.md (hunk 2 above)
Comment (mbwhite) [reply SUPPORT.md]: and I was looking for ages for the double 'the'... must be missing it..

Comment (awjh-ibm) [reply SUPPORT.md]: Think it should be worded "The npm module that are produced are" on second thoughts so

move the the

File: SUPPORT.md [hunk 5]
```diff
@@ -0,0 +1,49 @@
...
+Github is used for code base management, issues should reported in the [FABCN](https://jira.hyperledger.org/projects/FABCN/issues/) component in JIRA.
+
+
+## Summary of Compatibility
+
+Summary of the Fabric versions and the default node runtimes they use to host chaincode and contracts; along side the version of the node modules and which runtimes they are compatible with.
+
+|                       | Fabric 1.4 (node v8) | Fabric v2.0 (node v12) |
+| --------------------- | -------------------- | ---------------------- |
+| node modules **v1.4** | Yes                  | Yes                    |
```
Comment (heatherlp) [inline SUPPORT.md]: I still believe the node version should be a separate column/row in the table

Comment (awjh-ibm) [reply SUPPORT.md]: How linked are the node versions and fabric versions?

Comment (awjh-ibm) [reply SUPPORT.md]: Just don't want it to look like we support node 8 with v2 if v2 doesn;'t support it

Comment (mbwhite) [reply SUPPORT.md]: I've updated the table - @awjh-ibm @heatherlp @lesleyannjordan @davidkel ... do you want to see if that answers the question :-)

**Comment by heatherlp** [comment]
Hold on a minute - this PR is also adding support for node 12?!

**Comment by awjh-ibm** [comment]
@mbwhite Just to clarify the changes to the package.json files stating which version of node we support. We definitely only want this to state v12 not also v10?

**Comment by mbwhite** [comment]
> @mbwhite Just to clarify the changes to the package.json files stating which version of node we support. We definitely only want this to state v12 not also v10?

12 is the LTS and the only one that is part of the testing - the nodeenv image uses exactly the same one.. so yes I think just 12 is good.

**Pull request merged**